crates/cubecl-core/src/runtime_tests/mod.rs

```diff
@@ -17,21 +17,105 @@ pub mod unary;
 #[macro_export]
 macro_rules! testgen_all {
     () => {
-        use cubecl_core::prelude::*;
+        use $crate::Runtime;
 
-        cubecl_core::testgen_subcube!();
+        type FloatT = f32;
+        type IntT = i32;
+        type UintT = u32;
+
+        cubecl_core::testgen_assign!();
+        cubecl_core::testgen_branch!();
+        cubecl_core::testgen_const_match!();
+        cubecl_core::testgen_different_rank!();
         cubecl_core::testgen_launch!();
-        cubecl_core::testgen_cmma!();
-        cubecl_core::testgen_slice!();
+
+        $crate::testgen_untyped!();
+    };
+    ($f_def:ident: [$($float:ident),*], $i_def:ident: [$($int:ident),*], $u_def:ident: [$($uint:ident),*]) => {
+        use $crate::Runtime;
+
+        ::paste::paste! {
+            $(mod [<$float _ty>] {
+                type FloatT = $float;
+                type IntT = $i_def;
+                type UintT = $u_def;
```

Comment:
I would prefer `T` to be the full name. Is it `T` for template, type or test?

Reply:
type, I used the C++ convention I guess (`float32_t`). I can just spell it out I guess, it's not used in that many places